workers/task-driver/src/tasks/pay_fees.rs

```diff
@@ -0,0 +1,195 @@
+//! The `PayFees` task is responsible for settling the fees due for a given
+//! wallet
+
+// TODO: Remove this lint allowance
+#![allow(unused)]
+
+use std::{
+    error::Error,
+    fmt::{Display, Formatter, Result as FmtResult},
+};
+
+use arbitrum_client::client::ArbitrumClient;
+use async_trait::async_trait;
+use common::types::{tasks::PayFeesTaskDescriptor, wallet::WalletIdentifier};
+use job_types::proof_manager::ProofManagerQueue;
+use num_bigint::BigUint;
+use serde::Serialize;
+use state::{error::StateError, State};
+use tracing::instrument;
+
+use crate::{
+    driver::StateWrapper,
+    traits::{Task, TaskContext, TaskError, TaskState},
+};
+
+/// The name of the task
+const TASK_NAME: &str = "pay-fees";
+
+// --------------
+// | Task State |
+// --------------
+
+/// Defines the state of the fee payment task
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize)]
+pub enum PayFeesTaskState {
+    /// The task is awaiting scheduling
+    Pending,
+    /// The task is proving relayer fee payment for the ith balance
+    ProvingRelayerPayment,
+    /// The task is submitting a fee payment transaction for the ith balance
+    SubmittingPayment,
+    /// The task is proving protocol fee payment for the ith balance
+    ProvingProtocolPayment,
+    /// The task is submitting a fee payment transaction for the ith balance
+    SubmittingProtocolPayment,
+    /// The task is updating the validity proofs for the wallet
+    UpdatingValidityProofs,
+    /// The task has finished
+    Completed,
+}
+
+impl TaskState for PayFeesTaskState {
+    fn commit_point() -> Self {
+        PayFeesTaskState::ProvingRelayerPayment
```

Comment:
semantically, shouldn't this be either `SubmittingPayment` or `SubmittingProtocolPayment`? and since either of the variants could be the commit point, it seems to justify either splitting this out into two tasks, or having the proving/submitting state be agnostic of whether this is a relayer or protocol fee payment (and have that be captured in the task descriptor)

Reply:
Oops yeah, wrote this down before I defined all the states.

It's most convenient to make it one task per balance. This is the granularity at which fees are generated, so it makes caller logic straightforward.